### PR TITLE
Misc improvements and fixes

### LINKED ISSUE #299: Point data time looop
So a little issue I found.

The PointSource operation are separated from the main time loop in one case.

```
u[t, idx...] = u[t, idx...] + something
```
works good, but
```
u[t, idx...] = something
```
doesn't 

I created the branch `point-assign` to test it, `acoustic_example` is set up to use the new `assign` function that does `u[idx_rec] = rec`.

Sorry it is part of the wave-examples, I couldn't figure out how to add a trivial test, I have no idea what's going on in the `test_point_data.py`.

### LINKED ISSUE #304: dse/dle bug
So I created a branch called dse-dle-bug that highlights some of the current issues I found.

There is an example file called `bug-ex` in `examples/seismic/acoustic`
The corresponding operator is `forward_bug` in `operators`.
The kernel is the combination
```
stencils = [stencil(u), stencil(dux), stencil(duy)] #  3 different coupled PDE stencils
stencils += src.inject(u) # inject src in u
stencils += rec.interpolate(u) # read rec in u
```

Issues for the three stencils as `[stencilu, stencildux, stencilduy]` :
- dle=dse='advanced` => two last stencils removed from the code
- dse='noop, dle='advanced' => two last stencils  removed from the code
- dse='advanced, dle='noop' => two last stencils removed from the code
- dse='noop, dle='noop' => two last stencils  removed from the code

Issues for the three stencils as `[stencildux, stencilduy, stencilu]`  and  `[stencilduy, stencildux, stencilu]` :
- dse='noop, dle='noop' => OK
- dse='noop', dle='advanced` => dle fails and crashes
- dse='advanced', dle='noop` => OK
- dle=dse='advanced` => dle fails and crashes

### PR DESCRIPTION
**Pull request by FabioLuporini**
- A further DLE pass to minimise the negative impact of remainder loops when using 3D blocking
- Update benchmark.py  [depends on https://github.com/opesci/opescibench/pull/5]
- Fix a few issues, see below
- more tests

fixes #299 #304

### CODE REVIEW

**Comment by FabioLuporini** [comment]
Thoughts @mlange05 @navjotk  ?

File: tests/test_operator.py [hunk 1]
```diff
@@ -394,6 +394,33 @@ def test_consistency_perfect_loops(self, tu, tv, ti0, t0, t1):
...
+        ('Eq(ti0[x,y,z], ti0[x,y,z-1] + ti1[x,y,z+1])', 'Eq(ti1[x,y,z], ti3[x,y,z+1])',
+         'Eq(ti3[x,y,z], ti3[x,y,z-1] + ti0[x,y,z])'),
+        ('Eq(ti0[x,y,z+2], ti0[x,y,z-1] + ti1[x,y,z+1])',
+         'Eq(ti1[x,y,z+3], ti3[x,y,z+1])',
+         'Eq(ti3[x,y,z+2], ti0[x,y,z+1]*ti3[x,y,z-1])'),
+        ('Eq(ti0[x,y,z], ti0[x-2,y-1,z-1] + ti1[x+2,y+3,z+1])',
+         'Eq(ti1[x,y,z], ti3[x+1,y-4,z+1])',
+         'Eq(ti3[x,y,z], ti3[x+5,y,z-1] - ti0[x+3,y-2,z+4])')
+    ])
+    def test_consistency_coupled_w_ofs(self, exprs, ti0, ti1, ti3):
```
Comment (mlange05) [inline tests/test_operator.py]: Could we add a docstring to these two tests? They are a bit cryptic without at least a small description, I think.

Comment (FabioLuporini) [inline tests/test_operator.py]: Yes, will do

**Comment by FabioLuporini** [comment]
Merged.

**Comment by mloubout** [comment]
This needs to be reverted, then merge master. This is not compatible with the previously merged branch `3d-spacing`

**Comment by FabioLuporini** [comment]
It was passing all tests, so if there's an issue, it's in master

**Pull request merged**